### PR TITLE
Job Client: adds support for configuring the executor field on a job and populating the executor field on an instance

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds support for configuring the executor field on a job and populating the executor field on an instance

## Why are we making these changes?

This will enable the `JobClient` to specify the executor to use while submitting/loading a job. Additionally, it also allows displaying the executor used in an instance.

### CODE REVIEW

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Instance.java [hunk 1]
```diff
@@ -242,9 +258,11 @@ public String getHostName() {
     final private Boolean _preempted;
     final private String _outputURL;
     final private String _hostName;
+    private String _executor;
```
Comment (WenboZhao) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Instance.java]: Should we use Enum for this?

Comment (shamsimam) [reply jobclient/src/main/java/com/twosigma/cook/jobclient/Instance.java]: Will do.

Comment (shamsimam) [reply jobclient/src/main/java/com/twosigma/cook/jobclient/Instance.java]: Done.

**Comment by WenboZhao** [comment]
LGTM, feel free to merge. Or should I do that?

**Comment by shamsimam** [comment]
Please merge, I'll rebase the other PR then.

**Pull request merged**